data_display/views.py

```diff
@@ -81,6 +90,91 @@ def get_objects_by_table_and_sort(table_name, sort_by):
         'Not For Profits': (NotForProfits.objects.order_by(sort_by).values_list(), NotForProfits._meta.get_fields())
     }[table_name]
 
+def get_objects_by_table_and_filter(table_name, filter_table):
+    index = filter_table.find(',')
+    category = filter_table[:index]
+    value = filter_table[(index+1):]
+    #print(category)
+    #print(value)
+    if table_name == 'Students':
+        if category == 'student_id':
+            try:
+                return (Students.objects.filter(student_id=int(value)).values_list(), Students._meta.get_fields())
+            except:   # wrong input - return the default filter (name), which will be empty in this case
+                return (Students.objects.filter(name=value).values_list(), Students._meta.get_fields())
```

Comment:
Really shouldn't be using try catch statements for this. If anything that you call from Students.objects ends up having a bug that causes an Exception you'll never be able to tell since you're try-excepting all cases.

Just use an if else instead to check the type of `value` like so:

`if is_instance(value):`